docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,248 @@ If you need to have external labels in your query results, this is currently not
+   
+   ```bash
+   # Get list of all blocks in the bucket
+   thanos tools bucket inspect \
+       --objstore.config-file bucket.yaml \
+       --output=tsv > blocks.tsv
+       
+   # Find blocks from replica that we will drop    
+   cat blocks.tsv| grep prometheus_replica=<PROMETHEUS-REPLICA-TO-DROP> \
+       | awk '{print $1}' > blocks_to_drop.tsv
+       
+   # Mark found blocks for deletion
+   for ID in $(cat blocks_to_drop.tsv)
+   do
+       thanos tools bucket mark \
+          --marker="deletion-mark.json" \
+          --objstore.config-file bucket.yaml \
+          --details="Removed as duplicate" \
+          --id $ID
+   done
+   ```
+   
+   > **Note**: Replace **prometheus_replica** with the unique label that would differentiate prometheus replicas in your setup.
+
+   Again cleanup the duplicate blocks marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **ProTip!**: If you want to visualize what exactly is happening in the blocks with respect to the source of blocks, external labels, compaction levels, etc, you can use the below command to get the output as CSV and import it to a google sheet.
+   > ```bash 
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket-prod.yaml \
+   >     --output=csv > thanos-blocks.csv
+   > ```
+
+4) Relabel the blocks with external labels
+
+   Mimir doesn't inject external labels from the `meta.json` file into query results. Therefore we relabel the blocks with the required external labels in the `meta.json`. 
+   
+   > **Note**: You can get the external labels in the meta.json of each block from the CSV file imported in the above **ProTip** and build the rewrite config accordingly.
+   
+   Create a rewrite config like the following:
+   
+   ```bash
+   # relabel-config.yaml
+   - action: replace
+     target_label: "<LABEL-KEY>"
+     replacement: "<LABEL-VALUE"
+   ```
+   
+   Perform the rewrite dry run to confirm all work well.
+   
+   ```bash
+   # Get list of all blocks in the bucket after removing the depuplicate and downsampled blocks.
+   thanos tools bucket inspect \
+       --objstore.config-file bucket.yaml \
+       --output=tsv > blocks-to-rewrite.tsv
+   
+   # Check if rewrite of the blocks with external labels is working as expected.
+   for ID in $(cat blocks-to-rewrite.tsv)
+   do
+       thanos tools bucket rewrite \
+           --objstore.config-file bucket.yaml \
+           --rewrite.to-relabel-config-file relabel-config.yaml \
+           --dry-run \
+           --id $ID
+   done
+   ```
+   
+   After confirming that the rewrite is working as expected in **--dry-run**, you can apply the changes with the **--no-dry-run** flag. Don't forget to include **--delete-blocks**, otherwise the original blocks will not be marked for deletion.
```

Comment:
```suggestion
   After you confirm that the rewrite is working as expected via `--dry-run`, apply the changes with the `--no-dry-run` flag. Remember to include `--delete-blocks, otherwise the original blocks will not be marked for deletion.
```